probes/webhooksWithoutTokenAuth/def.yml

```diff
@@ -0,0 +1,33 @@
+# Copyright 2023 OpenSSF Scorecard Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+id: webhooksWithoutTokenAuth
+short: This check determines whether webhooks defined in the repository use a token to authenticate to remote servers.
+motivation: >
+  Webhooks without token authorization have the potential to make projects accessible to third-parties.
+implementation: >
+  The probe checks all webhooks of a project and checks whether each uses token authentication.
+outcome:
+  - If the project has one or more webhooks without token authorization, the probe returns as many OutcomeNegative (0) as the project has webhooks without token authorization.
+  - If the project does not have any webhooks without token authorization, the probe returns one OutcomePositive (1).
```

Comment:
Should it not return one finding per webhook? Each finding will return the metadata for a specific webhook, iiuc

Reply:
what do we have to uniquely identify a certain webhook? a URL?

Reply:
> what do we have to uniquely identify a certain webhook? a URL?

A path:

https://github.com/ossf/scorecard/blob/84bd607ae8c1b3d53c64172adcc0228e8f6ae298/clients/webhook.go#L18-L22